src/main/java/com/artipie/asto/streams/StorageValueAsStream.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/asto/LICENSE.txt
+ */
+package com.artipie.asto.streams;
+
+import com.artipie.asto.ArtipieIOException;
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.misc.UncheckedIOConsumer;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiConsumer;
+import org.cqfn.rio.Buffers;
+import org.cqfn.rio.WriteGreed;
+import org.cqfn.rio.stream.ReactiveInputStream;
+import org.cqfn.rio.stream.ReactiveOutputStream;
+
+/**
+ * Processes storage content as optional input and saves the result back as output stream.
+ * @since 1.5
+ */
+public final class StorageValueAsStream {
```

Comment:
The name looks a bit confusing because I'm expecting this class to be an implementation of `Stream`. Maybe we can choose a name to say that this class connects input and output, maybe something like `pipeline` or similar